framework/src/variables/MooseVariableFV.C

```diff
@@ -12,20 +12,53 @@
 #include "TimeIntegrator.h"
 #include "NonlinearSystemBase.h"
 #include "DisplacedSystem.h"
+#include "SystemBase.h"
+#include "SubProblem.h"
 #include "Assembly.h"
+#include "FVUtils.h"
+#include "FVFluxBC.h"
+#include "FVDirichletBC.h"
+
+#include "libmesh/numeric_vector.h"
 
 registerMooseObject("MooseApp", MooseVariableFVReal);
 
 template <typename OutputType>
 InputParameters
 MooseVariableFV<OutputType>::validParams()
 {
-  return MooseVariableField<OutputType>::validParams();
+  auto params = MooseVariableField<OutputType>::validParams();
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  params.template addParam<bool>("use_extended_stencil",
+                                 false,
+                                 "Whether to use an extended stencil for gradient computation.");
+#endif
+  return params;
 }
 
 template <typename OutputType>
 MooseVariableFV<OutputType>::MooseVariableFV(const InputParameters & parameters)
-  : MooseVariableField<OutputType>(parameters)
+  : MooseVariableField<OutputType>(parameters),
+    _solution(this->_sys.currentSolution()),
+    _phi(_assembly.template fePhi<OutputShape>(FEType(CONSTANT, MONOMIAL))),
+    _grad_phi(_assembly.template feGradPhi<OutputShape>(FEType(CONSTANT, MONOMIAL))),
+    _phi_face(_assembly.template fePhiFace<OutputShape>(FEType(CONSTANT, MONOMIAL))),
+    _grad_phi_face(_assembly.template feGradPhiFace<OutputShape>(FEType(CONSTANT, MONOMIAL))),
+    _phi_face_neighbor(
+        _assembly.template fePhiFaceNeighbor<OutputShape>(FEType(CONSTANT, MONOMIAL))),
+    _grad_phi_face_neighbor(
+        _assembly.template feGradPhiFaceNeighbor<OutputShape>(FEType(CONSTANT, MONOMIAL))),
+    _phi_neighbor(_assembly.template fePhiNeighbor<OutputShape>(FEType(CONSTANT, MONOMIAL))),
+    _grad_phi_neighbor(
+        _assembly.template feGradPhiNeighbor<OutputShape>(FEType(CONSTANT, MONOMIAL)))
```

Comment:
Because of all this stuff:
```
template <typename T>
void
Assembly::copyShapes(MooseVariableField<T> & v)
{
  phi(v).shallowCopy(v.phi());
  gradPhi(v).shallowCopy(v.gradPhi());
  if (v.computingSecond())
    secondPhi(v).shallowCopy(v.secondPhi());
}

void
Assembly::copyShapes(unsigned int var)
{
  auto & v = _sys.getVariable(_tid, var);
  if (v.fieldType() == Moose::VarFieldType::VAR_FIELD_STANDARD)
  {
    auto & v = _sys.getActualFieldVariable<Real>(_tid, var);
    copyShapes(v);
  }
  else if (v.fieldType() == Moose::VarFieldType::VAR_FIELD_ARRAY)
  {
    auto & v = _sys.getActualFieldVariable<RealEigenVector>(_tid, var);
    copyShapes(v);
  }
  else if (v.fieldType() == Moose::VarFieldType::VAR_FIELD_VECTOR)
  {
    auto & v = _sys.getActualFieldVariable<RealVectorValue>(_tid, var);
    copyShapes(v);
    if (v.computingCurl())
      curlPhi(v).shallowCopy(v.curlPhi());
  }
  else
    mooseError("Unsupported variable field type!");
}
```
Since I'm adding support for coupling FV variables into FE, then I need all these methods to be implemented and not error at run-time when called